File: Interview Preparation Kit - CPP/02. Arrays/005. Array Manipulation.cpp
// Problem: https://www.hackerrank.com/challenges/crush/problem
// Score: 60


#include <iostream>
#include <vector>
using namespace std;


int main(){
    int n, m;
    cin >> n >> m;

    vector<long long int> arr(n, 0);

    for (int i = 0; i < m; i++){
        int start, finish, value;
        cin >> start >> finish >> value;
        arr[start - 1] += value;
        arr[finish] -= value;
    }

    long long int ans = 0;
    long long int current = 0;

    for (int value: arr){
        current += value;
        if (current > ans){
            ans = current;
        }
    }

    cout << ans;

    return 0;
}

File: Interview Preparation Kit - Python/02. Arrays/005. Array Manipulation.py
# Problem: https://www.hackerrank.com/challenges/crush/problem
#  Score: 60


n, queries = map(int, input().split())

arr = [0 for i in range(n+2)]

for i in range(queries):
    start, finish, k = map(int, input().split())
    arr[start - 1] += k
    arr[finish] -= k

ans = 0
current = 0
for i in arr:
    current += i
    if current > ans:
        ans = current

print(ans)
